iac-extensions/arm/src/test/java/org/sonar/iac/arm/checks/PublicNetworkAccessCheckTest.java

```diff
@@ -19,19 +19,173 @@
  */
 package org.sonar.iac.arm.checks;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.stream.Stream;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
 
+import static org.junit.jupiter.params.provider.Arguments.of;
 import static org.sonar.iac.arm.checks.ArmVerifier.verify;
+import static org.sonar.iac.arm.checks.ArmVerifier.verifyContent;
 import static org.sonar.iac.common.testing.Verifier.issue;
 
 class PublicNetworkAccessCheckTest {
 
+  private static final PublicNetworkAccessCheck CHECK = new PublicNetworkAccessCheck();
+  private static final String MESSAGE_PUBLIC_NETWORK_ACCESS = "Make sure allowing public network access is safe here.";
+
   @Test
   void shouldCheckPublicNetworkAccess() {
     verify("PublicNetworkAccessCheckTest/Microsoft.Desktop_hostPools/test.json",
-      new PublicNetworkAccessCheck(),
-      issue(10, 31, 10, 40, "Make sure allowing public network access is safe here."),
+      CHECK,
+      issue(10, 31, 10, 40, MESSAGE_PUBLIC_NETWORK_ACCESS),
       issue(18, 31, 18, 59),
       issue(26, 31, 26, 54));
   }
+
+  static Stream<Arguments> shouldCheckPublicNetworkAccessSimplified() {
+    return Stream.of(
+      of("Microsoft.AgFoodPlatform/farmBeats", "2021-09-01-preview"),
+      of("Microsoft.ApiManagement/service", "2021-08-01"),
+      of("Microsoft.ApiManagement/service", "2021-04-01-preview"),
+      of("Microsoft.AppConfiguration/configurationStores", "2020-06-01"),
+      of("Microsoft.AppConfiguration/configurationStores", "2019-11-01-preview"),
```

Comment:
There is no test on the version, wondering why we are putting it in the test.

Reply:
I put them here to generate a valid, deployable input. If you think it doesn't make sense I can replace it by "dummy" apiVersion. WDYT?

Reply:
I would tend to remove it from there and directly inject your dummy apiVersion in the template if it has no impact on the rule detection logic.
As of now it seems we are injecting very specific version for each resource while in fact this is not used.
FYI on some rules we do have a check on the API version (Ex: only apply if version is 2021-10-01 or later), so that's why I was wondering here.